store/entity-store.html

```diff
@@ -285,25 +290,49 @@
 		removeListener: function(entityId, token, listener) {
 			return this._getToken(token).then(function (resolved) {
 				const cacheKey = resolved.cacheKey;
+				const tokenValue = resolved.tokenValue;
 
 				if (!entityId || typeof cacheKey !== 'string' || typeof listener !== 'function' || !this._listeners) {
 					return;
 				}
 
-				this._initContainer(this._listeners, entityId, cacheKey, new Set()).delete(listener);
+				const registrations = this._initContainer(this._listeners, entityId, cacheKey, new Map());
+
+				const tokenValues = registrations.get(listener);
+				if (!tokenValues) {
+					return;
+				}
+
+				// try to remove this specific registration, since a listener could be
+				// registered with multiple tokens (hopefully temporarily) despite sharing the same
+				// cache key (which is an internal detail)
+				if (!tokenValues.delete(tokenValue)) {
+					// we weren't aware of this particularly tokenValue for this listener, so
```

Comment:
Does it make this code simpler if we make `removeListener` a private method, so that clients are forced to use the function returned by `addListener` to remove listeners? I think that was originally my intention as I found that the original code copied from `siren-entity-mixins` was problematic when the token/href were changing. Not sure why I didn't make it private at that time. It looks like the `entity-loading-behavior` is calling `removeListener` directly, but maybe that could be refactored to use the same approach as `entity-behavior`.

Reply:
I don't know an exact path, but yeah probably something could be made simpler, because basically we'd want it to become

* call it once, remove it
* call it a second time, no-op

With this change, call it a second time becomes clear all listeners, which is still not super desirable

Reply:
I'll look into this to do as a breaking change later